File: flutter_app/test/widgets/care_header_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_app/widgets/care_header.dart';

void main() {
  group('CareHeader', () {
    testWidgets('displays title', (tester) async {
      await tester.pumpWidget(
        const MaterialApp(
          home: Scaffold(body: CareHeader(title: 'Test Title')),
        ),
      );
      expect(find.text('Test Title'), findsOneWidget);
    });

    testWidgets('displays subtitle when provided', (tester) async {
      await tester.pumpWidget(
        const MaterialApp(
          home: Scaffold(
            body: CareHeader(title: 'Title', subtitle: 'Sub'),
          ),
        ),
      );
      expect(find.text('Sub'), findsOneWidget);
    });

    testWidgets('shows back button when onBack provided', (tester) async {
      var pressed = false;
      await tester.pumpWidget(
        MaterialApp(
          home: Scaffold(
            body: CareHeader(title: 'Back', onBack: () => pressed = true),
          ),
        ),
      );
      expect(find.byIcon(Icons.chevron_left), findsOneWidget);
      await tester.tap(find.byIcon(Icons.chevron_left));
      expect(pressed, true);
    });

    testWidgets('hides back button when no onBack', (tester) async {
      await tester.pumpWidget(
        const MaterialApp(
          home: Scaffold(body: CareHeader(title: 'No Back')),
        ),
      );
      expect(find.byIcon(Icons.chevron_left), findsNothing);
    });

    testWidgets('shows accessibility button when onAccessibility provided', (tester) async {
      var pressed = false;
      await tester.pumpWidget(
        MaterialApp(
          home: Scaffold(
            body: CareHeader(
              title: 'A11y',
              onAccessibility: () => pressed = true,
            ),
          ),
        ),
      );
      expect(find.byIcon(Icons.accessibility_new), findsOneWidget);
      await tester.tap(find.byIcon(Icons.accessibility_new));
      expect(pressed, true);
    });
  });
}

File: react-native-app/src/components/__tests__/CareHeader.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import CareHeader from '../CareHeader';

describe('CareHeader', () => {
  test('renders title', () => {
    const { getByText } = render(<CareHeader title="Test Title" />);
    expect(getByText('Test Title')).toBeTruthy();
  });

  test('renders subtitle when provided', () => {
    const { getByText } = render(<CareHeader title="Title" subtitle="Sub" />);
    expect(getByText('Sub')).toBeTruthy();
  });

  test('shows back button when onBack provided', () => {
    const onBack = jest.fn();
    const { getByLabelText } = render(<CareHeader title="Title" onBack={onBack} />);
    fireEvent.press(getByLabelText('Go back'));
    expect(onBack).toHaveBeenCalledTimes(1);
  });

  test('hides back button when no onBack', () => {
    const { queryByLabelText } = render(<CareHeader title="Title" />);
    expect(queryByLabelText('Go back')).toBeNull();
  });

  test('shows emergency button when onEmergency provided', () => {
    const onEmergency = jest.fn();
    const { getByLabelText } = render(<CareHeader title="Title" onEmergency={onEmergency} />);
    fireEvent.press(getByLabelText('Emergency help'));
    expect(onEmergency).toHaveBeenCalledTimes(1);
  });
});
